File: docs/DbDiagram/Models/Incident.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DbDiagram.Models {
    public class Incident {
        public string Id { get; set; } = null!;

        public string Title { get; set; } = null!;

        public string Description { get; set; } = null!;

        public IncidentSeverity Severity { get; set; }

        public IncidentStatus Status { get; set; }

        public DateTime ReportedAt { get; set; }

        public virtual User ReportedBy { get; set; } = null!;

        public string ReportedById { get; set; } = null!;

        public DateTime? ResolvedAt { get; set; }

        public IncidentCategory Category { get; set; } = null!;

        public string CategoryId { get; set; } = null!;

        public double Latitude { get; set; }

        public double Longitude { get; set; }

        public string Location { get; set; } = null!;

        public virtual ICollection<IncidentActivity> Activities { get; set; } = new List<IncidentActivity>();
    }

    public enum IncidentStatus {
        Pending,        // Just reported, awaiting review or confirmation
        Investigating,  // Actively being looked into (by community or authorities)
        Resolved,       // Verified and successfully handled
        FalseAlarm      // Invalid report (mistake, spam, or irrelevant)
    }

    public enum IncidentSeverity {
        Low,
        Medium,
        High,
        Critical
    }
}

File: internal/models/incident.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type IncidentActivity struct {
	Id         string    `gorm:"primaryKey" json:"id"`
	IncidentId string    `json:"incidentId"`
	Incident   *Incident `json:"incident"`
	Message    string    `json:"message"`
	CreatedAt  time.Time `json:"createdAt"`
}

type Incident struct {
	CategoryId   string              `json:"categoryId"`
	Category     *Category           `gorm:"foreignKey:CategoryId;" json:"category"`
	Id           string              `gorm:"primaryKey" json:"id"`
	Code         string              `json:"code"`
	Summary      string              `json:"summary"`
	Severity     IncidentSeverity    `json:"severity"`
	Status       IncidentStatus      `json:"status"`
	UpdatedAt    time.Time           `json:"updatedAt"`
	ReportedAt   time.Time           `json:"reportedAt"`
	ReportedBy   *User               `json:"reportedBy"`
	ReportedById string              `json:"reportedById"`
	ResolvedAt   *time.Time          `json:"resolvedAt"`
	DeletedAt    gorm.DeletedAt      `gorm:"column:deleted_at;index" json:"deletedAt"`
	Latitude     float64             `json:"latitude"`
	Longitude    float64             `json:"longitude"`
	Location     string              `json:"location"`
	Activities   []*IncidentActivity `gorm:"foreignKey:IncidentId;" json:"activities"`
}

type IncidentStatus string

const (
	IncidentStatusPending       IncidentStatus = "pending"
	IncidentStatusInvestigating IncidentStatus = "investigating"
	IncidentStatusResolved      IncidentStatus = "resolved"
	IncidentStatusFalseAlarm    IncidentStatus = "falseAlarm"
)

type IncidentSeverity string

const (
	IncidentSeverityLow    IncidentSeverity = "low"
	IncidentSeverityMedium IncidentSeverity = "medium"
	IncidentSeverityHigh   IncidentSeverity = "high"
)
